demos/sample_actors/visual/sample_processor.py

```diff
@@ -0,0 +1,74 @@
+from improv.actor import Actor
+from queue import Empty
+import logging; logger = logging.getLogger(__name__)
+import zmq
+logger.setLevel(logging.INFO)
+import numpy as np
+import torch
+
+
+class Processor(Actor):
+    """
+    Process data and send it through zmq to be visualized.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def setup(self):
+        """
+        Creates and binds the socket for zmq.
+        """
+
+        self.name = "Processor"
+
+        gpu_available = torch.cuda.is_available()
+
+        if not gpu_available:
+            logger.error("GPU is needed for fastplotlib visualization.")
```

Comment:
If a GPU is available for `torch` it doesn't mean that everything required for WGPU is available. Also, limited software rendering on the CPU is possible with lavapipe.

This will work with wgpu>=0.13 :smile: 

```python
from wgpu.backends import wgpu_native

if len(wgpu_native.enumerate_adapters()) < 1:
  raise IOError("No WGPU compatible adapter found)"
```

Note that this will also include any lavapipe (CPU/software rendering) adapter that is available too!

The option to pick a specific GPU from fastplotlib is coming by the summer :D

Reply:
Actually, wouldn't you want to do this check in the visualization notebook itself? Since this actor isn't necessarily on the same computer as the visualization actor. Probably not necessary since `wgpu` will do the check anyways.

Reply:
latest fastplotlib does not import if no WGPU adapter is found: 

https://github.com/fastplotlib/fastplotlib/blob/41a694c632243eebdb204f4665bef49368c02781/fastplotlib/__init__.py#L20-L21

Also GPU info and selection will become easier after https://github.com/fastplotlib/fastplotlib/pull/491 thanks to Almar's work :smile: